infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LogReplicationEventListener.java

```diff
@@ -59,10 +59,6 @@ public void onNext(CorfuStreamEntries results) {
 
         // If the current node is not a leader, ignore the notifications.
         synchronized (discoveryService) {
-            if (!discoveryService.getSessionManager().getReplicationContext().getIsLeader().get()) {
```

Comment:
last minor comment, I thought we no longer need to synchronize on discoveryService? since the `input` method is synchronized already

Reply:
yes, that's right.  Removed it.  

Actually, I think that we dont need 'synchronized' on the input() method as well because there is a single instance of DiscoveryService.  But I have not kept it as it is right now.